src/symbolize/mod.rs

```diff
@@ -117,6 +117,29 @@ pub use symbolizer::Symbolizer;
 use crate::Addr;
 
 
+/// A enumeration of the different input types the symbolization APIs
+/// support.
+#[derive(Clone, Copy, Debug)]
+pub enum Input<A, O> {
+    /// A virtual address.
+    ///
+    /// A virtual address is an address as a process would see it, for
+    /// example. A virtual address may contain relocation or address space
+    /// randomization effects.
+    VirtAddr(A),
+    /// A file address.
+    ///
+    /// A file address is an address as it would appear in a binary or
+    /// debug symbol file, i.e., one excluding any relocations.
+    FileAddr(A),
+    /// A file offset.
+    ///
+    /// A file offset is the linear offset of a byte in a file in the file
+    /// system from its start.
+    FileOffset(O),
```

Comment:
tbh, I find FileAddr confusing. My suggestion would be:

VirtAddr -> AbsAddr (absolute address) or perhaps just Addr
FileAddr -> VirtOffset (perhaps could be VirtAddr to match ELF terminology, I guess, though with shared libraries and PIE executables "address" is a lie and it is actually offset within virtual address space).
FileOffset -> FileOffset

From user space perspective there is no "virtual" address, it's all real/absolute addresses for them. It's only kernel that knows about physical and virtual addresses. For me it's more about offsets vs absolute addresses, and then we get two kinds of offsets, one file contents and another is, for lack of better term, virtual offset.